backend/src/routers/user.ts

```diff
@@ -93,4 +95,49 @@ userRouter.get(
     })
 );
 
+userRouter.get(
+    '/settings/',
+    expressAsyncHandler(async (req, res) => {
+        const payload = await readToken(req);
+        if (payload === null) {
+            res.status(401).send('No valid token on the request found');
+            return;
+        }
+
+        const data = await selectUserSettings(payload.userID);
```

Comment:
Add error handling for this function call or change the behaviour of the function.

Reply:
readToken already verifies the user is valid, the only way selectUserSettings would throw error if database connection is broken or the user is deleted and theres a lingering invalid jwt token.